pkg/util/templates/templates.go

```diff
@@ -86,9 +91,18 @@ type multiWindowMultiBurnAlertRule struct {
 			 targeturl: getmeright
 */
 func (r *multiWindowMultiBurnAlertRule) render(url, percent, label, alertName string) monitoringv1.Rule {
-	alertString := "1-avg_over_time(probe_success{" + label + "}[" + r.shortWindow + "]) > (" + r.burnRate + "*(1-" + percent + "))\n" +
-		"and\n" +
-		"1-avg_over_time(probe_success{" + label + "}[" + r.longWindow + "]) > (" + r.burnRate + "*(1-" + percent + " ))"
+	sWindow, _ := time.ParseDuration(r.shortWindow)
+	lWindow, _ := time.ParseDuration(r.longWindow)
+	mPeriod, _ := time.ParseDuration(serviceMonitorPeriod)
+
+	alertString := "1-(sum(sum_over_time(probe_success{" + label + "}[" + r.shortWindow + "])) / sum(count_over_time(probe_success{" + label + "}[" + r.shortWindow + "])))" + 
+	"> (" + r.burnRate + "*(1-" + percent + "))\n" + 
+	"and sum(count_over_time(probe_success{" + label + "}[" + r.shortWindow + "])) > " + fmt.Sprintf("%d", int(sWindow.Minutes()/mPeriod.Minutes()/2)) +
+	"\nand\n" +
+	"1-(sum(sum_over_time(probe_success{" + label + "}[" + r.longWindow + "])) / sum(count_over_time(probe_success{" + label + "}[" + r.longWindow + "])))" + 
+	"> (" + r.burnRate + "*(1-" + percent + "))\n" + 
+	"and sum(count_over_time(probe_success{" + label + "}[" + r.longWindow + "])) > " + fmt.Sprintf("%d", int(lWindow.Minutes()/mPeriod.Minutes()/2))
```

Comment:
for future us: can you try to make this a bit more readable/understandable?

Reply:
Would you like me to refactor it?

Reply:
yes please